challenges.yml

```diff
@@ -193,3 +193,245 @@ stimulus_values:
+    }
+    ```
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          The `clickOutside()` method should be prefixed with `async`
+        b: |
+          The `useClickOutside` feature must be initialized in the `connect()` method
+        c: |
+          The controller is missing a target.
+        d: |
+          Nothing.
+      correct: b
+multi_controller_dispatch:
+  question: |
+    How can I pass the `id` of an item from one to another Stimulus controller?
+  explanation: |
+    The `stimulus-use` library comes with the `useDispatch` behavior which helps with dispatching and listening to events.
+    Dispatching an event is as easy as `this.dispatch('event:name');` and, if you specify its second argument, then
+    it will be passed to its listeners.
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          You cannot communicate between controllers.
+        b: |
+          By dispatching an event as `this.dispatch('event:name', null, item.id);`
+        c: |
+          By dispatching an event as `this.dispatch('event:name', item.id);`
+        d: |
+          By defining a global variable.
+      correct: c
+react_props:
+  question: |
+    Select the right option for rendering a React component with data coming from a template in a Stimulus controller
+  explanation: |
+    You can use the `Values` API of Stimulus to receive data from a template and then pass it
+    through a React component as a property field.
+  features:
+    multiple_choice:
+      choices:
+        a: |
+          ```js
+          export default class extends Controller {
+              static targets = [
+                  'name'
+              ]
+
+              connect() {
+                  ReactDOM.render(
+                      <Product name={this.nameTarget} />,
+                      this.element
+                  )
+              }
+          }
+          ```
+        b: |
+          ```js
+          export default class extends Controller {
+              connect(productName) {
+                  ReactDOM.render(
+                      <Product name={productName} />,
+                      this.element
+                  )
+              }
+          }
+          ```
+        c: |
+          ```js
+          export default class extends Controller {
+              static values = {
+                  name: String
+              }
+
+              connect() {
+                  ReactDOM.render(
+                      <Product name={this.nameValue} />,
+                      this.element
+                  )
+              }
+          }
+          ```
+      correct: c
+async_imports:
+  question: |
+    What's the right option for importing dynamically the `ReactDOM` module in a Stimulus controller?
+  explanation: |
+    Webpack allows you to do dynamic imports by following the next syntax:
+
+    ```js
+    import('react-dom').then((ReactDOM) => {
+        // ReactDOM.default.render(...);
+    }
+    ```
```

Comment:
Then, below this, show the equivalent await syntax.

Heck, I think you could even get hipster with some object destructuring!!!

```
const { default: ReactDOM } = await import('react-dom').
```

That... would require some testing 😆